cli/cli/cmds_volume.go

```diff
@@ -339,7 +344,11 @@ func (c *CLI) initVolumeCmds() {
 			// volumes that are attached or can be attached should be valid
 			// candidates for a mount operation
 			result, err := c.lsVolumes(
-				args, volumeStatusAttached, volumeStatusAvailable)
+				args,
+				requestAttachments(
+					apitypes.VolumeAttachmentsMine,
+					apitypes.VolumeAttachmentsAttached,
+					apitypes.VolumeAttachmentsUnattached))
```

Comment:
Hi @codenrhoden,

I got cut off above (hit enter too soon). I may be totally off base above, but can you see what I was attempting with the above mask?

One thing to note, and in regards to something you said:

> Why use the bitmask at all, then, instead of just `Requested`?

Because to do any of the mapping to an instance or device mapping requires that clients send in the request headers the instance ID and local device information. The mask provides the ability to retrieve volumes without also requiring the presence of the instance ID and/or local device information in the headers.

Previously I assumed drivers would check to see if the instance ID was present, and if so, automatically return volume attachment information for only that instance. 

However that was too constricting as it didn't leave us the ability to create a client that automatically handles sending the instance ID for a user. Additionally, it meant for some drivers we could not request volumes that are both attached to a provided instance but also available (unattached to any other instance). 

See what I mean?

Or am I totally missing something obvious here? I could be. What are your thoughts?

FWIW, see [these tests](https://github.com/akutz/libstorage/blame/bugfix/attachment-filtering-logic/drivers/storage/vfs/tests/vfs_test.go#L166-L312) I added to showcase the various ways to use the mask. Each test asserts the return values are what they are expected to be.